### DISCUSSION TITLE
Handling forbidden combinations with `except*`

### DISCUSSION THREAD

**Discussion by jakkdl** [microsoft/pyright#9409] [Ideas]
I'm not sure how pyright wants to deal with syntax errors in general, but pyright currently doesn't flag any of the forbidden combinations from PEP-654 https://peps.python.org/pep-0654/#forbidden-combinations (mypy flags all of them)

There's probably some scenarios where this might matter for typing, and in case users aren't running other linters it might be good if pyright notifies the user immediately.

**Comment by erictraut** [comment]
I agree that pyright should report these as syntax errors, consistent with the runtime. I'll convert this to an enhancement request.

**Reply by erictraut** [reply]
Here's a [link](https://github.com/microsoft/pyright/issues/9416) to the enhancement request for reference.
